shared/AppInsightsCore/src/JavaScriptSDK/BaseCore.ts

```diff
@@ -249,14 +265,182 @@ export class BaseCore implements IAppInsightsCore {
             };
 
             _self.releaseQueue = () => {
-                if (_eventQueue.length > 0) {
-                    arrForEach(_eventQueue, (event: ITelemetryItem) => {
+                if (_isInitialized && _eventQueue.length > 0) {
+                    let eventQueue = _eventQueue;
+                    _eventQueue = [];
+
+                    arrForEach(eventQueue, (event: ITelemetryItem) => {
                         _self.getProcessTelContext().processNext(event);
                     });
-
-                    _eventQueue = [];
                 }
             };
+
+            /**
+             * Periodically check logger.queue for log messages to be flushed
+             */
+             _self.pollInternalLogs = (eventName?: string): number => {
+                 _internalLogsEventName = eventName;
+
+                let interval = _self.config.diagnosticLogInterval;
+                if (!interval || !(interval > 0)) {
+                    interval = 10000;
+                }
+                if(_internalLogPoller) {
+                    clearInterval(_internalLogPoller);
+                }
+                _internalLogPoller = setInterval(() => {
+                    _flushInternalLogs();
+                }, interval) as any;
+
+                return _internalLogPoller;
+            }
+
+            /**
+             * Stop polling log messages from logger.queue
+             */
+            _self.stopPollingInternalLogs = (): void => {
+                if(_internalLogPoller) {
+                    clearInterval(_internalLogPoller);
+                    _internalLogPoller = 0;
+                    _flushInternalLogs();
+                }
+            }
+
+            // Add addTelemetryInitializer
+            proxyFunctions(_self, () => _telemetryInitializerPlugin, [ "addTelemetryInitializer" ]);
```

Comment:
This effectively "adds" the `addTelemetryInitializer` function to this instance and as we are within the `dynamicProto` section it will also become a prototype level function.

This helper `proxyFunctions` avoids a lot of the boilerplate code needed to "forward" the function call and arguments using something like

```ts
_self.addTelemetryInitializer = function(telemetryInitializer: TelemetryInitializerFunction) {
    _telemetryInitializerPlugin.addTelemetryInitializer(telemetryInitializer);
}
```